src/olympia/constants/search.py

```diff
@@ -81,6 +82,7 @@
     'hungarian': 'light_hungarian',
     'indonesian': 'indonesian',
     'italian': 'light_italian',
+    'punjabi': 'punjabi',
```

Comment:
Unfortunately ElasticSearch doesn't seem to support a specific stemmer for Punjabi https://www.elastic.co/guide/en/elasticsearch/reference/5.3/analysis-stemmer-tokenfilter.html so you'll have to omit this line.